next/pages/form/[slug]/[[...id]].tsx

```diff
@@ -0,0 +1,141 @@
+import { FormDefinition } from '@backend/forms/types'
+import { getFormDefinition } from '@backend/utils/forms'
+import { formsApi } from '@clients/forms'
+import { GetFileResponseDto } from '@clients/openapi-forms'
+import { RJSFSchema, UiSchema } from '@rjsf/utils'
+import { GetServerSidePropsContext } from 'next'
+import { useRouter } from 'next/router'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+
+import { FormStateProvider } from '../../../components/forms/FormStateProvider'
+import GeneratedFormRJSF from '../../../components/forms/GeneratedFormRJSF'
+import { FormFileUploadStateProvider } from '../../../components/forms/useFormFileUpload'
+import AccountPageLayout from '../../../components/layouts/AccountPageLayout'
+import PageWrapper from '../../../components/layouts/PageWrapper'
+import {
+  getSSRAccessToken,
+  GetSSRCurrentAuth,
+  getSSRCurrentAuth,
+  ServerSideAuthProviderHOC,
+} from '../../../components/logic/ServerSideAuthProvider'
+import { environment } from '../../../environment'
+import { InitialFormData } from '../../../frontend/types/initialFormData'
+import logger from '../../../frontend/utils/logger'
+
+type FormTestPageProps = {
+  schema: RJSFSchema
+  uiSchema: UiSchema
+  page: { locale: string }
+  initialFormData: InitialFormData
+  ssrCurrentAuthProps: GetSSRCurrentAuth
+}
+
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  if (!environment.featureToggles.forms) return { notFound: true }
+
+  // TODO: Remove and support non-auth version of the page
+  const ssrCurrentAuthProps = await getSSRCurrentAuth(ctx.req)
+  if (!ssrCurrentAuthProps.userData) {
+    return {
+      redirect: {
+        destination: '/prihlasenie',
+        permanent: false,
+      },
+    }
+  }
+
+  let formDefinition: FormDefinition
+  try {
+    formDefinition = getFormDefinition(ctx.query.slug)
+  } catch (error) {
+    logger.error(error)
+    return { notFound: true }
+  }
+
+  const hasIdQueryParam = Array.isArray(ctx.query.id)
+
+  // It is not possible to have only one optional route parameter ([[id]]), so we have to check if it is an array and
+  // if it has only one element.
+  if (hasIdQueryParam && (ctx.query.id as string[]).length !== 1) {
```

Comment:
Using `ctx.params` instead of `ctx.query` returns typed params instead of `any`.

That's also what we usually use :) 

https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#context-parameter

Reply:
Perfect, that's what I needed.